minecode/miners/golang.py

```diff
@@ -237,3 +237,31 @@ def build_golang_package(package_data, purl):
         vcs_url=vcs_url,
     )
     return package
+
+
+def build_golang_generic_package(package_data, package_url):
+    """Return a single Golang package"""
+    homepage_url = "/".join(["https:/", package_url.namespace, package_url.name])
+    license_text = package_data.get("licenses")
+    extracted_license_statement = [license_text]
+
+    download_url = (
```

Comment:
Go has some weird rules to encode upper case in these strings. See https://github.com/aboutcode-org/go-inspector/blob/442bc5b83d5aeff2b7a27937ec82b63277bc8f7c/src/go_inspector/utils.py#L211 

We are adding support for getting golang download URL in PURL library. @pombredanne @chinyeungli I think we can reuse that here ?

Reply:
https://github.com/package-url/packageurl-python/pull/195 here is PR for same.

Reply:
@TG1999 Thanks. We can use the code there once it's merged.